assets/tests/hooks/useNotifications.test.tsx

```diff
@@ -131,29 +137,80 @@ describe("useNotifications", () => {
       () => {
         useNotifications(mockAddNotification, mockSetNotifications)
       },
-      { wrapper: wrapper(mockSocket, []) }
+      { wrapper, initialProps: { socket: mockSocket, selectedRouteIds: [] } }
     )
 
     expect(reloadSpy).toHaveBeenCalled()
     reloadSpy.mockRestore()
   })
-})
 
-const wrapper =
-  (socket: Socket | undefined, selectedRouteIds: RouteId[]) =>
-  ({ children }: { children: ReactElement<HTMLElement> }) =>
-    (
-      <SocketProvider
-        socketStatus={{ socket, connectionStatus: ConnectionStatus.Connected }}
-      >
-        <StateDispatchProvider
-          state={{
-            ...initialState,
-            routeTabs: [routeTabFactory.build({ selectedRouteIds })],
-          }}
-          dispatch={jest.fn()}
-        >
-          {children}
-        </StateDispatchProvider>
-      </SocketProvider>
+  test("doesn't rejoin channel on every render", () => {
+    const mockAddNotification = jest.fn()
+    const mockSetNotifications = jest.fn()
+    const mockSocket = makeMockSocket()
+    const mockChannel = makeMockChannel("ok", {
+      initial_notifications: ["notification1", "notification2"],
+    })
+    mockSocket.channel.mockImplementation(() => mockChannel)
+
+    const { rerender } = renderHook(
+      () => {
+        useNotifications(mockAddNotification, mockSetNotifications)
+      },
+      {
+        wrapper,
+        initialProps: { socket: mockSocket, selectedRouteIds: ["route"] },
+      }
     )
+    rerender({ socket: mockSocket, selectedRouteIds: ["route"] })
+
+    expect(mockChannel.join).toHaveBeenCalledTimes(1)
+  })
+
+  test("rejoins channel if selected routes change", () => {
+    const mockAddNotification = jest.fn()
+    const mockSetNotifications = jest.fn()
+    const mockSocket = makeMockSocket()
+    const mockChannel = makeMockChannel("ok", {
+      initial_notifications: ["notification1", "notification2"],
```

Comment:
**nitpick:** Is there any reason to include the notifications here? Could you just repsond `[]`? (Same for above.)

Reply:
I guess not strictly necessary, I can remove them.